webrtc/RTCPeerConnection-getStats.html

```diff
@@ -0,0 +1,187 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>RTCPeerConnection.prototype.getStats</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="RTCPeerConnection-helper.js"></script>
+<script src="dictionary-helper.js"></script>
+<script src="RTCStats-helper.js"></script>
+<script>
+  'use strict';
+
+  // Test is based on the following editor draft:
+  // https://w3c.github.io/webrtc-pc/archives/20170605/webrtc.html
+  // https://w3c.github.io/webrtc-stats/archives/20170614/webrtc-stats.html
+
+  // The following helper function is called from RTCPeerConnection-helper.js
+  //   getTrackFromUserMedia
+
+  // The following helper function is called from RTCStats-helper.js
+  //   validateStatsReport
+  //   assert_stats_report_has_stats
+
+  /*
+    8.2.  RTCPeerConnection Interface Extensions
+      partial interface RTCPeerConnection {
+        Promise<RTCStatsReport> getStats(optional MediaStreamTrack? selector = null);
+      };
+
+    8.3.  RTCStatsReport Object
+      interface RTCStatsReport {
+        readonly maplike<DOMString, object>;
+      };
+
+    8.4.  RTCStats Dictionary
+      dictionary RTCStats {
+        DOMHighResTimeStamp timestamp;
+        RTCStatsType        type;
+        DOMString           id;
+      };
+
+      id
+        Two RTCStats objects, extracted from two different RTCStatsReport objects, MUST
+        have the same id if they were produced by inspecting the same underlying object.
+
+    8.2.  getStats
+      1.  Let selectorArg be the method's first argument.
+      2.  Let connection be the RTCPeerConnection object on which the method was invoked.
+      3.  If selectorArg is neither null nor a valid MediaStreamTrack, return a promise
+          rejected with a newly created TypeError.
```

Comment:
I skipped this one because it has been covered by WebIDL and shouldn't reach the step. Has filed w3c/webrtc-pc#1498 for that.